tests/unit_tests/output_parsers/test_base_output_parser.py

```diff
@@ -0,0 +1,43 @@
+"""Test the BaseOutputParser class and its sub-classes."""
+from abc import ABC
+from typing import List, Set, Type, Optional
+from unittest.mock import MagicMock
+
+import pytest
+
+from langchain.schema import BaseOutputParser
+
+_PARSERS_TO_SKIP = {"FakeOutputParser", "BaseOutputParser"}
+
+
+def non_abstract_subclasses(cls: Type[ABC], to_skip: Optional[Set]=None) -> List[Type]:
+    """Recursively find all non-abstract subclasses of a class."""
+    subclasses = []
+    for subclass in cls.__subclasses__():
+        if not getattr(subclass, "__abstractmethods__", None):
+            if subclass.__name__ not in to_skip:
+                subclasses.append(subclass)
+        subclasses.extend(non_abstract_subclasses(subclass, to_skip))
+    return subclasses
+
+
+@pytest.mark.parametrize("cls", non_abstract_subclasses(BaseOutputParser, _PARSERS_TO_SKIP))
+def test_all_subclasses_implement_type(cls: Type[BaseOutputParser]) -> None:
```

Comment:
very small nit: test_subclass_implements_type is slightly more appropriate name given this asserts over a single class at a time